src/spac/spatial_analysis.py

```diff
@@ -120,25 +159,75 @@ def Cluster_Interaction_Matrix_Analysis(
         fig = plt.figure()
         ax = fig.add_subplot(1, 1, 1)
 
+    # Operational Section
+    # -----------------------------------------------
+
     # Create a categorical column data for plotting
     new_annotation_name = annotation + "_plot"
 
     adata.obs[new_annotation_name] = pd.Categorical(
         adata.obs[annotation])
 
+    if stratify_by:
+        if isinstance(stratify_by, list):
+            adata.obs[
+                'concatenated_obs'

```

Comment:
@ruiheesi, suggestion: save the string "concatenated_obs" in a variable and use that variable later in the code.

Reply:
The concatenated_obs is an obs stored in adata.obs, do you mean saving it to another one? or do you want the string itself also be a variable?